src/libraries/Common/src/System/IO/Compression/ZLibNative.cs

```diff
@@ -173,6 +174,61 @@ public ZLibStreamHandle()
                 SetHandle(IntPtr.Zero);
             }
 
+            public static ZLibStreamHandle CreateForDeflate(CompressionLevel level, int windowBits,
+                int memLevel, CompressionStrategy strategy)
+            {
+                ZLibStreamHandle zLibStreamHandle = new ZLibStreamHandle();
+                ErrorCode errC;
+
+                try
+                {
+                    errC = zLibStreamHandle.DeflateInit2_(level, windowBits, memLevel, strategy);
+                }
+                catch (Exception cause) // could not load the ZLib dll
```

Comment:
Nit: cause is an unusual name for an exception. We typically use `e`, `ex`, or something like that.